### PR TITLE
18zoo sr powers

### PR DESCRIPTION
**Pull request by carlorusso**

### CODE REVIEW

File: lib/engine/game/g_18_zoo/game.rb [hunk 1]
```diff
@@ -1125,6 +1129,10 @@ def it_s_all_greek_to_me
           @it_s_all_greek_to_me ||= company_by_id('IT_S_ALL_GREEK_TO_ME')
         end
 
+        def it_s_all_greek_to_me_active?
```
Comment (jenf) [inline lib/engine/game/g_18_zoo/game.rb:1132]: it_is

Comment (carlorusso) [reply lib/engine/game/g_18_zoo/game.rb:1132]: what do you mean? the power name is "IT'S ALL GREEK TO ME"
maybe "it_s_all_greek_to_me_is_active?"

Comment (jenf) [reply lib/engine/game/g_18_zoo/game.rb:1132]: the power name might be, but it_s reads badly, it_is_all_greek_to_me_active? is readable

Comment (carlorusso) [reply lib/engine/game/g_18_zoo/game.rb:1132]: I can simplify to greek_to_me too

Comment (carlorusso) [reply lib/engine/game/g_18_zoo/game.rb]: done in f3d4a846

File: lib/engine/game/g_18_zoo/game.rb [hunk 2]
```diff
@@ -1125,6 +1129,10 @@ def it_s_all_greek_to_me
           @it_s_all_greek_to_me ||= company_by_id('IT_S_ALL_GREEK_TO_ME')
         end
 
+        def it_is_all_greek_to_me_active?
+          it_s_all_greek_to_me.all_abilities.any? { |ability| ability.is_a?(Engine::Ability::Close) }
```
Comment (tobymao) [inline lib/engine/game/g_18_zoo/game.rb:1133]: not consisntent

File: lib/engine/game/g_18_zoo/round/operating.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+module Engine
+  module Game
+    module G18ZOO
+      module Round
+        class Operating < Engine::Round::Operating
+          def after_setup
+            super
+
+            @game.corporations.each do |corporation|
+              corporation.all_abilities
```
Comment (tobymao) [inline lib/engine/game/g_18_zoo/round/operating.rb:15]: one liner

corporation.all_albities.each do |ability|  

next if ...

File: lib/engine/game/g_18_zoo/step/choose_power.rb [hunk 4]
```diff
@@ -11,10 +11,21 @@ def choice_name
...
+          @choices[:midas] = 'Midas' if company_midas?(entity)
+          if company_holiday?(entity)
+            @game.corporations.select(&:ipoed).each do |corporation|
+              @choices["holiday:#{corporation.id}"] = "Holiday (#{corporation.id})"
+            end
+          end
+          @choices[:greek_to_me] = 'It’s all greek to me' if company_greek?(entity)
+          if company_whatsup?(entity)
+            train = @game.depot.depot_trains&.first
+            entity.presidencies
```
Comment (tobymao) [inline lib/engine/game/g_18_zoo/step/choose_power.rb:23]: one liner

File: lib/engine/game/g_18_zoo/round/operating.rb (hunk 3 above)
Comment (carlorusso) [reply lib/engine/game/g_18_zoo/round/operating.rb]: done in 5d6145b5

File: lib/engine/game/g_18_zoo/step/choose_power.rb (hunk 4 above)
Comment (carlorusso) [reply lib/engine/game/g_18_zoo/step/choose_power.rb]: like this?
```suggestion
            entity.presidencies.each do |corporation|
              next if corporation.cash < train&.price
              @choices["whatsup:#{corporation.id}:#{train.id}"] = "Whatsup (#{corporation.id}) (#{train.name})"
            end
```

Comment (carlorusso) [reply lib/engine/game/g_18_zoo/step/choose_power.rb]: is 5d6145b5 ok?

File: lib/engine/game/g_18_zoo/game.rb (hunk 2 above)
Comment (carlorusso) [reply lib/engine/game/g_18_zoo/game.rb]: done in 5d6145b5

**Pull request merged**